packages/babel-helper-module-transforms/src/index.ts

```diff
@@ -61,7 +61,13 @@ export interface RewriteModuleStatementsAndPrepareHeaderOptions {
   loose?: boolean;
   importInterop?: ImportInterop;
   noInterop?: boolean;
-  lazy?: Lazy;
+  lazy?: Lazy.Lazy;
+  getWrapperPayload?: (
+    source: string,
+    metadata: SourceModuleMetadata,
+    importNodes: t.Node[],
+  ) => unknown;
+  wrapReference?: (ref: t.Expression, payload: unknown) => t.Expression | null;
```

Comment:
Nit: can we make this interface generic? So that TS knows that the `payload` passed into `wrapReference` must conforms to type of the return of `getWrapperPayload`.

Reply:
Unfortunately, `payload` might be unknown because it might be generated by the hook of another plugin. The best we can do is
```js
interface T {
  wrapReference?: (ref: t.Expression, payload: T | unknown) => t.Expression | null;
}
```
which does not improve types (`T | unknown` is `unknown`) but it might improve autocompletion.

Reply:
> it might be generated by the hook of another plugin

Good point. That seems to be similar to the `PluginPass`'s behaviour: Any plugin can augment the `PluginPass` but there is no guarantee that the `PluginPass` must not be changed by other plugins within the same pass.